native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/tasks/BuildNativeImageTask.java

```diff
@@ -240,19 +240,18 @@ public void exec() {
         }
     }
 
-    private void checkRequiredVersionIfNeeded(File executablePath, NativeImageOptions options) {
-        if (!options.getRequiredVersion().isPresent()) {
-            return;
-        }
+    private int checkVersionAndGetMajorJDKVersion(File executablePath, NativeImageOptions options) {
```

Comment:
suggestion: Can you make this a public method in `NativeImageUtils` (you'll probably need to make `NativeImageUtils` public too)? Otherwise I will have to reimplement the very same logic in the Micronaut Gradle plugin.

Reply:
This already is a public method of the public `NativeImageUtils` :)

Reply:
Sorry I'm confused. In `NativeImageUtils` I can see `getMajorJDKVersion` but not the whole `checkVersionAndGetMajorJDKVersion` which will need to be replicated in Micronaut.